examples/neural_quantum_states/src/models/alt_transformer.py

```diff
@@ -0,0 +1,168 @@
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from .base import Base
+
+class AltTransformer(Base):
+    def __init__(self, num_sites: int, num_spin_up: int, num_spin_down: int, embedding_dim: int=16, nhead: int=2, dim_feedforward: int=64, num_layers: int=1, temperature: float=1.0, device: str=None, **kwargs):
+        '''
+        A Transformer-based autoregressive NQS Ansatz using the phase strategy from Bennewitz et al, where a single linear layer operators on the concated list of transformer hidden states in lieu of a seperate phase network.
+        Child class specific args:
+            embedding_dim: dimension of transformer hidden states
+            nhead: number of attention heads
+            dim_feedforward: dimension of transformer feedforward layer
+            num_layers: number of transformer blocks
+            temperature: modulus network softmax temperature parameter
+            device: device to store model on
+        '''
+        super(AltTransformer, self).__init__('AltTransformer', num_sites, num_spin_up, num_spin_down, device)
+
+        # construct model
+        self.num_in, self.num_out = num_sites, num_sites*2
+        self.temperature = temperature
+        self.input_order = np.stack([np.arange(self.num_sites-2,-1,-2), np.arange(self.num_sites-1,-1,-2)],1).reshape(-1) # [4,5,2,3,0,1]
+        self.input_order = torch.Tensor(self.input_order).int().to(self.device)
+        self.shell_order = torch.arange(self.num_sites//2-1, -1, -1) # [2,1,0]
+        
+        transformer_layer = nn.TransformerEncoderLayer(embedding_dim, nhead, dim_feedforward=dim_feedforward, dropout=0.0, batch_first=True)
```

Comment:
A comment above for this block of code would help identify what's this code block for.